infra/lambda/lambda_function.py

```
@@ -338,6 +407,64 @@ def require_groups(session: Dict[str, Any], scanner_type: str) -> None:
 
 
+def _http_request_audit_context(event: Dict[str, Any]) -> Dict[str, Any]:
+    """Correlation and client context from API Gateway for audit logs."""
+    ctx: Dict[str, Any] = {}
+    rc = event.get('requestContext') or {}
+    if not isinstance(rc, dict):
+        return ctx
+    ctx['request_id'] = rc.get('requestId')
+    http = rc.get('http') or {}
+    if isinstance(http, dict):
+        if http.get('sourceIp'): ctx['source_ip'] = http.get('sourceIp')
```

Ruff (0.15.9)

[error] Multiple statements on one line (colon)

(E701)